### PR TITLE
lilypond-unstable: drop because lilypond is newer and it does not build

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -357,6 +357,7 @@ mapAliases ({
   libsexy = throw "libsexy has been removed from nixpkgs, as it's abandoned and no package needed it."; # 2019-12-10
   libstdcxxHook = throw "libstdcxx hook has been removed because cc-wrapper is now directly aware of the c++ standard library intended to be used."; # 2020-06-22
   libqmatrixclient = throw "libqmatrixclient was renamed to libquotient"; # added 2020-04-09
+  lilypond-unstable = lilypond; # added 2021-03-11
```
Comment (Mindavi) [inline pkgs/top-level/aliases.nix:360]: For backwards compat?

**Comment by Mindavi** [comment]
Makes sense to me.

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/top-level/aliases.nix:360]: Yes. This prevents eval errors and upgrades people to the latest version.

**Comment by yurrriq** [comment]
imo we want to maintain stable (even minor versions) and unstable (odd minor versions), just like upstream. #184327 gets us back on track with unstable at 2.23.11

**Comment by Mindavi** [comment]
Sure, as long as it's maintained that's fine. I'd say this PR was just cleaning up an unmaintained derivation

**Comment by SuperSandro2000** [comment]
> imo we want to maintain stable (even minor versions) and unstable (odd minor versions), just like upstream

If upstream releases often enough we don't need this. Just use the odd even unstable something updater and keep the stable package.

**Pull request merged**